src/tpm2_wrap.c

```diff
@@ -7029,837 +7029,323 @@ int wolfTPM2_GetPolicyDigest(TPM_HANDLE sessionHandle, byte* policyDigest,
-
-    if (rc == 0) {
-        /* add PCR to the policy digest */
-        rc = wolfTPM2_PolicyPCR(session->handle.hndl, pcrAlg, pcrArray,
-            pcrArraySz);
-    }
-
-    if (rc == 0) {
-        /* add PolicySigned to the policy */
-        /* set the key handle */
-        policySignedIn->authObject = authKey->handle.hndl;
-
-        /* set the session */
-        policySignedIn->policySession = session->handle.hndl;
-
-        /* set nonce if it's non null */
-        if (nonceSz > 0) {
-            policySignedIn->policyRef.size = nonceSz;
-            XMEMCPY(policySignedIn->policyRef.buffer, nonce, nonceSz);
-        }
-
-        /* set auth sigAlg */
-        if (authKey->pub.publicArea.type == TPM_ALG_ECC) {
-            policySignedIn->auth.sigAlg =
-                authKey->pub.publicArea.parameters.eccDetail.scheme.scheme;
-
-            if (policySignedIn->auth.sigAlg == TPM_ALG_NULL)
-                policySignedIn->auth.sigAlg = TPM_ALG_ECDSA;
-        }
-        else if (authKey->pub.publicArea.type == TPM_ALG_RSA) {
-            policySignedIn->auth.sigAlg =
-                authKey->pub.publicArea.parameters.rsaDetail.scheme.scheme;
-
-            if (policySignedIn->auth.sigAlg == TPM_ALG_NULL)
-                policySignedIn->auth.sigAlg = TPM_ALG_RSASSA;
-        }
-
-        /* set hash alg */
-        policySignedIn->auth.signature.any.hashAlg = pcrAlg;
-
-        /* set auth signature */
-        switch (authKey->pub.publicArea.type) {
-            case TPM_ALG_ECC:
-                policySignedIn->auth.signature.ecdsa.signatureR.size =
-                    policySignedSigSz / 2;
-                XMEMCPY(policySignedIn->auth.signature.ecdsa.signatureR.buffer,
-                    policySignedSig,
-                    policySignedIn->auth.signature.ecdsa.signatureR.size);
-
-                policySignedIn->auth.signature.ecdsa.signatureS.size =
-                    policySignedSigSz / 2;
-                XMEMCPY(policySignedIn->auth.signature.ecdsa.signatureS.buffer,
-                    policySignedSig + policySignedSigSz / 2,
-                    policySignedIn->auth.signature.ecdsa.signatureS.size);
-                break;
-            case TPM_ALG_RSA:
-                policySignedIn->auth.signature.rsassa.sig.size =
-                    policySignedSigSz;
-                XMEMCPY(policySignedIn->auth.signature.rsassa.sig.buffer,
-                    policySignedSig, policySignedSigSz);
-                break;
-            default:
-                rc = BAD_FUNC_ARG;
-                break;
-        }
-    }
-
-    if (rc == 0) {
-        rc = TPM2_PolicySigned(policySignedIn, policySignedOut);
-    }
-
-    /* get the policyDigest */
-    if (rc == 0) {
-        rc = wolfTPM2_GetPolicyDigest(session->handle.hndl, policyDigest,
-            &policyDigestSz);
-    }
+    inSz = *digestSz; /* capture input digest size (for policyDigestOld) */
+    rc = TPM2_GetHashType(pcrAlg);
```

Comment:
why set rc if you're going to overwrite it before checking?

Reply:
Haven't found a better way to solve this. Same pattern is used elsewhere. Resolves a compiler warning about int to enum.

Reply:
sorry I didn't see hashType was being set to it, which is fine. I thought it was being overwritten by 7163